### PR TITLE
Add nvlist payload stats for zdb -ll dump

### PR DESCRIPTION
**Pull request by don-brady**
### Description
When dumping the ZFS vdev lable with 'zdb -ll', also dump some nvlist stats to help analyze the current footprint.

### Motivation and Context
Recent discussions on adding additional nvlist data for dRAID to the vdev lable prompted collecting this data. Over time, additional name value pairs, like `vdev_enc_sysfs_path` have been added. This change
helps to determine the closeness to full capacity for vdev labels.

A sample dump output looks like:
```
ZFS Label NVList Config Stats:
  6000 bytes used, 108648 bytes free (using  5.2%)

   integers:   58   2104 bytes (35.07%)
    strings:   53   3492 bytes (58.20%)
   booleans:    2     92 bytes ( 1.53%)
    nvlists:   13    312 bytes ( 5.20%)

 leaf vdevs:   10    490 bytes average
                     500 bytes largest
```
### How Has This Been Tested?
tested with zdb(1) and ztest(1) on centos 7.2, x86_64

### Types of changes
- [X] New feature (non-breaking change which adds functionality)

### Checklist:
- [X] My code follows the ZFS on Linux code style requirements.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

File: cmd/zdb/zdb.c [hunk 1]
```diff
@@ -2235,6 +2235,136 @@ dump_label_uberblocks(vdev_label_t *lbl, uint64_t ashift)
...
+dump_nvlist_stats(nvlist_t *nvl, size_t cap)
+{
+	zdb_nvl_stats_t stats = { 0 };
+	size_t size, sum = 0, total;
+	size_t noise, average;
+
+	/* requires nvlist with non-unique names for stat collection */
+	VERIFY0(nvlist_alloc(&stats.zns_string, 0, 0));
+	VERIFY0(nvlist_alloc(&stats.zns_uint64, 0, 0));
+	VERIFY0(nvlist_alloc(&stats.zns_boolean, 0, 0));
```
Comment (behlendorf) [inline cmd/zdb/zdb.c]: `fnvlist_alloc()` was added for this kind of usage.

File: cmd/zdb/zdb.c [hunk 2]
```diff
@@ -2292,6 +2422,8 @@ dump_label(const char *dev)
 			nvlist_t *vdev_tree = NULL;
 
 			dump_nvlist(config, 4);
+			if (l == 3 && dump_opt['l'] >= 2)
```
Comment (behlendorf) [inline cmd/zdb/zdb.c]: Not a real problem, but the `l == 3` check which makes sense now will likely cause a tiny conflict down the road.  @richardelling made the excellent suggestion of updating `zdb -l` such that it only dumps one label when they're all the same.

File: cmd/zdb/zdb.c (hunk 1 above)
Comment (don-brady) [inline cmd/zdb/zdb.c]: Understood, but we explicitly need non-unique names here and the `fnvlist_alloc` variant is always `NV_UNIQUE_NAME`

File: cmd/zdb/zdb.c (hunk 2 above)
Comment (don-brady) [inline cmd/zdb/zdb.c]: Yeah, good point.  I could  clone the first config nvlist encountered and post process it to get the desired effect of having this stat dump be last.  Is that acceptable?

File: cmd/zdb/zdb.c (hunk 1 above)
Comment (behlendorf) [inline cmd/zdb/zdb.c]: Ahh I see, I didn't pick up on that initially.  Good point.

File: cmd/zdb/zdb.c (hunk 2 above)
Comment (behlendorf) [inline cmd/zdb/zdb.c]: I don't think that's necessary.  I just wanted to make note of this potential issue.  I know @ofaaland was looking in to implementing this.

**Pull request merged**